src/components/DappBrowser/Homepage.tsx

```diff
@@ -178,16 +195,16 @@ const Card = React.memo(function Card({
     const dappUrl = site.url;
     const iconUrl = site.image;
     const host = getDappHost(dappUrl);
-    const overrideFound = dapps.find(dapp => dapp.url === host);
+    const overrideFound = dapps.find((dapp: { url: string }) => dapp.url === host);
     if (overrideFound?.iconUrl) {
       return overrideFound.iconUrl;
     }
     return iconUrl;
-  }, [site.image, site.url]);
+  }, [dapps, site.image, site.url]);
 
   return (
     <Box>
-      <ButtonPressAnimation onPress={() => goToUrl(site.url)} scaleTo={0.94}>
+      <ButtonPressAnimation onPress={() => handlePress()} scaleTo={0.94}>
```

Comment:
you should replace this arrow function with just `handlePress`. by having an arrow function here you're needlessly defining another function on top of handlePress, and since it is passed as a prop it will get recreated every render. not a big deal here at all, but just letting you know bc with more complex functions it can add significant overhead